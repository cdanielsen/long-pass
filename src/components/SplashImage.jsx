import React from 'react'
import PropTypes from 'prop-types'

function SplashImage ({id, imageUrl}) {
  return (
    <a href="https://xkcd.com/936/">
      <img id={id} src={imageUrl} />
    </a>
  )
}

SplashImage.propTypes = {
  imageUrl: PropTypes.string,
  id: PropTypes.string
}

export default SplashImage
