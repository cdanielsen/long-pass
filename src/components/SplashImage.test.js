import React from 'react'
import SplashImage from './SplashImage.jsx'
import { shallow } from 'enzyme'

describe('SplashImage component', () => {
  test('Should return an image tag with an id and the correct src attribute', () => {
    const imagePath = 'http://image.com'
    const id = 'sploosh'
    const wrapper = shallow(<SplashImage id={id} imageUrl={imagePath} />)

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.is('a')).toBe(true)
    expect(wrapper.find('img').prop('src')).toEqual(imagePath)
    expect(wrapper.find('img').prop('id')).toEqual(id)
  })
})
