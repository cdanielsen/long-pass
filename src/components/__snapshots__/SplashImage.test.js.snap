// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashImage component Should return an image tag with an id and the correct src attribute 1`] = `
<a
  href="https://xkcd.com/936/"
>
  <img
    id="sploosh"
    src="http://image.com"
  />
</a>
`;
